x-pack/plugins/endpoint/server/routes/alerts/list/types.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { decode, encode } from 'rison-node';
+import { LoggerFactory } from 'kibana/server';
+import { esFilters, TimeRange } from '../../../../src/plugins/data/common';
+import { JsonObject } from '../../../../src/plugins/data/common/es_query/kuery';
+import { EndpointConfigType } from './config';
+
+/**
+ * Request params for alert queries.
+ *
+ * Must match exactly the values that the API receives.
+ */
+export interface AlertListRequestParams {
+  page_index?: number;
+  page_size?: number;
+  filters?: string;
+  query?: string;
+  sort?: string;
+  order?: string;
+  after?: string;
+  before?: string;
+}
+
+/**
+ * Request metadata for additional context.
+ *
+ * Internal use: contains the validated request parameters for use
+ * by the application. Keys are camel-cased and do not necessarily
+ * match the names of the request parameters that were passed in.
+ */
+export class AlertListRequestData {
+  [key: string]: any;
+  pageSize!: number;
+  pageIndex?: number;
+  fromIndex?: number;
+  query?: string;
+  sort!: string;
+  order!: string;
+  searchAfter?: [any, any];
+  searchBefore?: [any, any];
+  next?: string;
+  prev?: string;
+
+  // Rison-encoded
+  private _filters?: esFilters.Filter[];
+  private _dateRange?: TimeRange;
+
+  public get filters(): esFilters.Filter[] {
+    return this._filters;
+  }
+
+  public set filters(_filters: any) {
```

Comment:
We should be more strict with the type here

Reply:
nit: I think the `_` prefix on parameters also indicates to the linter that the variable is unused, might be worth changing 🤷‍♂